server/server_test.go

```diff
@@ -213,6 +213,42 @@ func TestGetConnectURLs(t *testing.T) {
 	}
 }
 
+func TestClientAdvertiseConnectURL(t *testing.T) {
+	opts := DefaultOptions()
+	opts.Port = 4222
+	opts.ClientAdvertise = "nats.example.com"
+	s := New(opts)
+	defer s.Shutdown()
+
+	urls := s.getClientConnectURLs()
```

Comment:
You are unit testing `getClientConnectURLs()`, which is good, but in this case you are not testing ClientAdvertise per se. Like I described above, this function is not called when there is no cluster defined, so clients would still receive default host/port, not the one configured in client advertise. Is that ok?